src/test/ui/issue-35675.stderr

```diff
@@ -0,0 +1,65 @@
+error[E0412]: cannot find type `Apple` in this scope
+  --> $DIR/issue-35675.rs:20:29
+   |
+20 | fn should_return_fruit() -> Apple {
+   |                             ^^^^^
+   |                             |
+   |                             not found in this scope
+   |                             help: did you mean the variant's enum `Fruit`
+
+error[E0425]: cannot find function `Apple` in this scope
+  --> $DIR/issue-35675.rs:23:5
+   |
+23 |     Apple(5)
+   |     ^^^^^ not found in this scope
+   |
+help: possible candidate is found in another module, you can import it into scope
+   | use Fruit::Apple;
+
+error[E0573]: expected type, found variant `Fruit::Apple`
+  --> $DIR/issue-35675.rs:28:33
+   |
+28 | fn should_return_fruit_too() -> Fruit::Apple {
+   |                                 ^^^^^^^^^^^^
+   |                                 |
+   |                                 not a type
+   |                                 help: did you mean the variant's enum `Fruit`
+
+error[E0425]: cannot find function `Apple` in this scope
+  --> $DIR/issue-35675.rs:31:5
+   |
+31 |     Apple(5)
+   |     ^^^^^ not found in this scope
+   |
+help: possible candidate is found in another module, you can import it into scope
+   | use Fruit::Apple;
+
+error[E0573]: expected type, found variant `Ok`
+  --> $DIR/issue-35675.rs:36:13
+   |
+36 | fn foo() -> Ok {
+   |             ^^ not a type
+   |
+   = help: there is an enum variant `std::prelude::v1::Ok`,did you mean to use `std::prelude::v1`?
+   = help: there is an enum variant `std::prelude::v1::Result::Ok`,did you mean to use `std::prelude::v1::Result`?
```

Comment:
IIRC, this test was moved to compile-fail due to instability in the order of suggestions (see https://github.com/rust-lang/rust/pull/40775), so this will probably have to do be reverted, unless you want to fix the instability.